src/integrationTests/async.spec.ts

```diff
@@ -175,3 +195,165 @@ describe('when window.__uid2.init is called on SdkLoaded from a callback', () =>
     expect(sdkWindow.__uid2.getAdvertisingToken()).toBe(identity.advertising_token);
   });
 });
+
+describe('when getFreshAdvertisingToken is called before init complete', () => {
+  describe('when initialising with a valid identity', () => {
+    const identity = makeIdentity();
+    test('it should resolve promise after invoking the callback', () => {
+      const p = uid2.getFreshAdvertisingToken().then((token: any) => {
+        expect(callback).toHaveBeenCalled();
+        return token;
+      });
+      uid2.init({ callback: callback, identity: identity });
+      jest.runAllTimers();
+      return expect(p).resolves.toBe(identity.advertising_token);
+    });
+  });
+
+  describe('when initialising with an invalid identity', () => {
+    test('it should reject promise after invoking the callback', () => {
+      const p = uid2.getFreshAdvertisingToken().catch((e: any) => {
+        expect(callback).toHaveBeenCalled();
+        throw e;
+      });
+      uid2.init({ callback: callback });
+      return expect(p).rejects.toBeInstanceOf(Error);
+    });
+  });
+
+  describe('when initalising with a non-expired identity which requires a refresh', () => {
+    test('it should resolve with new advertising token after refresh', () => {
+      const originalIdentity = makeIdentity({
+        refresh_from: Date.now() - 100000,
+      });
+      const updatedIdentity = makeIdentity({
+        advertising_token: 'updated_advertising_token'
+      });
+      const p = uid2.getFreshAdvertisingToken().then((token: any) => {
+        expect(callback).toHaveBeenCalled();
```

Comment:
Callback doesn't seem to have anything to do with this test. I think you can achieve the same result just with `const p = uid2.getFreshAdvertisingToken()`;